code/modules/atmospherics/machinery/other/miner.dm

```diff
@@ -132,8 +138,10 @@
 	if(!isopenturf(O))
 		return FALSE
 	var/datum/gas_mixture/merger = new
-	merger.assert_gas(spawn_id)
-	merger.gases[spawn_id][MOLES] = (spawn_mol)
+	var/moles_spawned = moles_to_spawn(O)
+	for(var/gas in gas_mixture)
+		merger.assert_gas(gas)
+		merger.gases[gas][MOLES] = moles_spawned * gas_mixture[gas]
```

Comment:
*\**?????

Reply:
Gas mixture is an associative list, that maps gas type to relative ratio,  for example, the airmix is `list("n2" = N2STANDARD, "o2" = O2STANDARD)`,

Thus If moles_to_spawn() is 5, it will add about 4 moles of N2, and 1 mole of O2.

I agree, the syntax is a bit confusing, especially since when you do an enhanced for loop over an associative list, it iterates over the keys rather than the values of the list, but that's a byond quirk.

I'll make a commit that clarifies this in a comment.